src/generate/utils/execute/property/OperationExecutor_Observation.cpp

```diff
@@ -0,0 +1,36 @@
+//
+// Created by Carolin on 13.06.2020.
+//
+
+#include <util/GameLogicUtils.hpp>
+#include "../OperationExecutor.hpp"
+
+std::vector<spy::gameplay::State_AI>
+OperationExecutor::executeObservation(const spy::gameplay::State_AI &state, const spy::gameplay::PropertyAction &op,
+                                      const spy::MatchConfig &config, const libclient::LibClient &libClient) {
+    spy::gameplay::State_AI sSuccess = state;
+
+    auto sourceChar = state.getCharacters().findByUUID(op.getCharacterId());
+    auto targetChar = spy::util::GameLogicUtils::findInCharacterSetByCoordinates(state.getCharacters(), op.getTarget());
+
+    if (sourceChar->getFaction() == targetChar->getFaction() ||
+        libClient.getUnknownFactionList().find(targetChar->getCharacterId()) ==
+        libClient.getUnknownFactionList().end() || config.getObservationSuccessChance() == 0) {
+        return {}; // action is not useful (no new information will be gained) -> do not continue branch
```

Comment:
Ich würde erwarten dass eine valide aktion mindestens einen result-state hat, auch wenn sie unsinnig ist.

Reply:
Generell ja, aber wenn ich die unsinnigen Sachen gleich "wegschmeiße" spare ich mir die Kinder / Folgezustände von der unsinnigen Aktion, die tendentiell wieder Folgezustände haben. Und zudem muss ich nicht nochmal in der evaluations-Funktion herausfinden, ob das unsinnig ist. Denn dort würde ich das dann ja so oder so "wegschmeißen".
Also schmeiße ich das gleich weg und spare mir unnötigen Rechenaufwand.